containers/src/Kokkos_DualView.hpp

```diff
@@ -501,6 +511,24 @@ template< class , class , class , class > friend class DualView ;
     }
   }
 
+  void sync_host() {
+    if( ! std::is_same< typename traits::data_type , typename traits::non_const_data_type>::value )
+      Impl::throw_runtime_exception("Calling sync_host on a DualView with a const datatype.");
+    if(modified_flags.data()==NULL) return;
+    if(modified_flags(1) > modified_flags(0)) {
+      deep_copy (h_view, d_view);
+    }
+  }
+
+  void sync_device() {
+    if( ! std::is_same< typename traits::data_type , typename traits::non_const_data_type>::value )
+      Impl::throw_runtime_exception("Calling sync_device on a DualView with a const datatype.");
+    if(modified_flags.data()==NULL) return;
+    if(modified_flags(0) > modified_flags(1)) {
+      deep_copy (d_view, h_view);
```

Comment:
@crtrott @mhoemmen I think the `modified_flags` need to be reset s.t. `modified_flags(1) == modified_flags(0)` after the `deep_copy`; let me know if I shouldn't push those changes if the behavior was intended to not reset them equal for some reason.